domain/externalcontroller/state/state.go

```diff
@@ -181,3 +182,85 @@ VALUES (?, ?)
 
 	return errors.Trace(err)
 }
+
+func (st *State) ImportExternalControllers(ctx context.Context, infos []externalcontroller.MigrationControllerInfo) error {
```

Comment:
I think we decided to just get import to call `UpdateExternalController` instead. As that does the upsert as well.

Reply:
see [comment](https://github.com/juju/juju/pull/15838#discussion_r1257538461)